### PR TITLE
Fix common name of SSL cert

### PR DESCRIPTION
**Pull request by matthewrmshin**
Use suite host name instead of suite name.

### CODE REVIEW

File: lib/cylc/suite_srv_files_mgr.py [hunk 1]
```diff
@@ -464,11 +464,11 @@ def _get_ssl_cert(self, path, reg, pkey_obj):
             # OpenSSL not installed, so we can't use HTTPS anyway.
             return
         # Use suite name as the 'common name', but no more than 64 chars.
```
Comment (hjoliver) [inline lib/cylc/suite_srv_files_mgr.py]: (comment now wrong?)

Comment (matthewrmshin) [inline lib/cylc/suite_srv_files_mgr.py]: Yes. Now fixed.

**Comment by hjoliver** [comment]
ssl.com FAQ "What is the common name": http://info.ssl.com/article.aspx?id=10048

**Comment by hjoliver** [comment]
The code looks fine.  I suppose we don't need to worry about the problem (see FAQ URL above) with variant versions of the same host, because `get_suite_host()` always supplies the same host name that will also be used by tasks comms?   Presumably the common name be a raw IP address if necessary?

**Comment by matthewrmshin** [comment]
Yes. Clients will always use the host value in `~/cylc-run/SUITE/.service/contact` to communicate with the suite server, and the host value in the file is set using `get_suite_host`. The raw IP address method is covered by `get_suite_host` as well, so we should not have to worry.

**Pull request merged**